### DISCUSSION TITLE
How can I use a collection with the method graphQLResponse.dataAsObject?

### DISCUSSION THREAD

**Discussion by jmbarroso** [Netflix/dgs-framework#163] [Q&A] [answered]
Hi all,

I'm struggling to use the method `graphQLResponse.dataAsObject(..)` described in the [doc](https://netflix.github.io/dgs/advanced/java-client/) when the response contains a collection.  

The below sentence fails on compilation time with the error _Only classes are allowed on the left-hand side of a class literal_:
`val employers : Collection<Employee> = graphQLResponse.dataAsObject(List<Employee>::class.java)`

Note that my generated classes are Kotlin. 

I've tried multiple variations, but I cannot make it work. Any hint on what is going on and how to make it work?

Thank you!

**Comment by paulbakker** [comment, marked as answer]
Use the `extractValueAsObject(path: String, typeRef: TypeRef<T>)` version.
For example:

```kotlin
 val listOfSubmittedBy: List<String> = graphQLResponse.extractValueAsObject("submitReview.edges[*].node.submittedBy", object: TypeRef<List<String>>() {})
```

**Reply by paulbakker** [reply]
And you can drop the explicit type declaration as well.

**Reply by jmbarroso** [reply]
Worked. Thanks!

**Reply by paulbakker** [reply]
Added a test case to the framework as well, I found that we didn't have that. 
https://github.com/Netflix/dgs-framework/pull/164
